src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -707,14 +707,21 @@ uct_rc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
                                   const uct_iface_is_reachable_params_t *params)
 {
     uint8_t my_type = uct_rc_mlx5_iface_get_address_type(tl_iface);
+    uint8_t remote_type;
     const uct_iface_addr_t *iface_addr;
 
     iface_addr = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD, params,
                                  iface_addr, IFACE_ADDR, NULL);
+    remote_type = *(uint8_t*)iface_addr;
     /* Check hardware tag matching compatibility */
-    if ((iface_addr != NULL) && (my_type != *(uint8_t*)iface_addr)) {
-        uct_iface_populate_info_str_buf(
-                            params, "incompatible hardware tag matching");
+    if ((iface_addr != NULL) && (my_type != remote_type)) {
+        UCT_IFACE_FILL_INFO_STR_BUF(
+                    params, "incompatible hardware tag matching. "
+                    "%s (local) vs %s (remote)",
+                    my_type == UCT_RC_MLX5_IFACE_ADDR_TYPE_TM ?
+                                        "tag matching" : "basic",
+                    remote_type == UCT_RC_MLX5_IFACE_ADDR_TYPE_TM ?
+                                        "tag matching" : "basic");
```

Comment:
1. use ( ) around == expression
2. can we unite common code in these 2 arguments?

Reply:
2. I guess only with using a map such `static const tm_type_to_str[2] = {"basic", "tag matching"};` but it requires protecting it before accessing etc. What do you think?

Reply:
I can at least reduce it to one comparison. I guess it may make it more efficient, this way
```
(my_type == UCT_RC_MLX5_IFACE_ADDR_TYPE_TM) ?
                    "tag matching" : "basic",
(my_type == UCT_RC_MLX5_IFACE_ADDR_TYPE_TM) ?
                    "basic" : "tag matching");
```

Reply:
maybe it can go without protection, since the value has to be 0 or 1 in a way that cannot be affected by user input?